health.go

```diff
@@ -194,18 +194,10 @@ func (h *Health) Measure(ctx context.Context) Check {
 			}()
 
 			resCh := make(chan error)
-			defer close(resCh)
 
 			go func() {
-				res := c.Check(ctx)
-
-				select {
-				case <-resCh:
-					// channel is already closed
-					return
-				default:
-					resCh <- res
-				}
+				resCh <- c.Check(ctx)
+				defer close(resCh)
```

Comment:
@lucasmdrs , the defer should be before the Check, no? if the Check does not end, this will never get closed?

Reply:
Hi @grzn 
Changing the order of the defer would not change anything, endless checks will always result in leaky goroutines.

The changes done here will make open channels not be closed until those goroutines are finished.
It's still not the best solution, but ensures that there's no race condition between timeouts and checks

I'll work to improve this entire logic a bit, but that will be out of the scope for the issue